File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/PractitionerRole.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class PractitionerRole : DomainResource, IResource
{
    public PractitionerRole_AvailableTime[]? AvailableTime { get; set; }
    public Base.CodeableConcept[]? Specialty { get; set; }
    public PractitionerRole_NotAvailable[]? NotAvailable { get; set; }
    public Base.ResourceReference? Organization { get; set; }
    public bool? Active { get; set; }
    public Base.CodeableConcept[]? Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public string? AvailabilityExceptions { get; set; }
    public Base.ResourceReference? Practitioner { get; set; }
    public Base.ContactPoint[]? Telecom { get; set; }
    public Base.Period? Period { get; set; }
    public Base.ResourceReference[]? Location { get; set; }
    public Base.ResourceReference[]? Endpoint { get; set; }
    public Base.ResourceReference[]? HealthcareService { get; set; }

    public class PractitionerRole_AvailableTime : BackboneElement
    {
        public bool? AllDay { get; set; }
        public string[]? DaysOfWeek { get; set; }
        public string? AvailableEndTime { get; set; }
        public string? AvailableStartTime { get; set; }
    }

    public class PractitionerRole_NotAvailable : BackboneElement
    {
        public Base.Period? During { get; set; }
        public required string Description { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/PractitionerRole.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import CodeableConcept
from base import ContactPoint
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class PractitionerRoleAvailableTime(BackboneElement):
    all_day: Optional[bool] = None
    days_of_week: Optional[List[str]] = field(default_factory=list)
    available_end_time: Optional[str] = None
    available_start_time: Optional[str] = None

@dataclass(kw_only=True)
class PractitionerRoleNotAvailable(BackboneElement):
    during: Optional[Period] = None
    description: str

@dataclass(kw_only=True)
class PractitionerRole(DomainResource):
    available_time: Optional[List[PractitionerRoleAvailableTime]] = field(default_factory=list)
    specialty: Optional[List[CodeableConcept]] = field(default_factory=list)
    not_available: Optional[List[PractitionerRoleNotAvailable]] = field(default_factory=list)
    organization: Optional[Reference] = None
    active: Optional[bool] = None
    code: Optional[List[CodeableConcept]] = field(default_factory=list)
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    availability_exceptions: Optional[str] = None
    practitioner: Optional[Reference] = None
    telecom: Optional[List[ContactPoint]] = field(default_factory=list)
    period: Optional[Period] = None
    location: Optional[List[Reference]] = field(default_factory=list)
    endpoint: Optional[List[Reference]] = field(default_factory=list)
    healthcare_service: Optional[List[Reference]] = field(default_factory=list)